1/numberDivisors.js

```diff
@@ -0,0 +1,24 @@
+// complexity is O(log n) - I'm not sure
+function numberDivisors(number) {
+  const result = [];
+  for (let divisor = 1; divisor < (number/divisor + 1); divisor++) {
+    if (number % divisor === 0 && !result[divisor]) { result.push(divisor); result.push(number/divisor);}
+  }
+  return result;
+}
+
+
+
+
+console.log(numberDivisors(6)); // [1,6,2,3]
+console.log(numberDivisors(1)); // [1, 1] a side effect for 1
```

Comment:
All numbers that are squares will have this effect.
_For example:_ for input 6, the result will be: [1, 36, 2, 18, 3, 12, 4, 9, 6, 6].
There should be no duplicate numbers.

Reply:
thnks, fixed